back-nest/src/sсrapers/scraper-rozetka.service.ts

```diff
@@ -1,16 +1,24 @@
 import { Injectable, Logger } from '@nestjs/common';
 import puppeteer from 'puppeteer';
 import * as cheerio from 'cheerio';
+import { randomUUID } from 'crypto';
 
 import { Sources } from './models/sources';
 import { ScrapedProduct } from './models/scraped-product.model';
+import { ProductService } from '../products/products.services';
+import { ScraperService } from './models/scraper-service.model';
+import { S3Service } from '../s3/s3.service';
 import { RedisService } from '../redis/redis.service';
 
 @Injectable()
-export class ScraperRozetkaService {
+export class ScraperRozetkaService implements ScraperService {
   private readonly logger = new Logger(ScraperRozetkaService.name);
 
-  constructor(private readonly redisService: RedisService) {}
+  constructor(
+    private readonly redisService: RedisService,
+    private readonly productService: ProductService,
+    private readonly s3Service: S3Service,
+  ) {}
 
   async scrapeRozetkaProducts(): Promise<ScrapedProduct[]> {
```

Comment:
You don't need to include provider into this method's name, because the scope is rozetka already.
```suggestion
  async scrapeProducts(): Promise<ScrapedProduct[]> {
```